### ISSUE TITLE
1.15.2: NoSuchMethodError

### ISSUE THREAD

**Issue by XaaREl** [kangarko/ChatControl#904]
Minecraft Version: 1.15.2
MySQL: false
BungeeCord: false

Hi. I'm getting errors like this:

On startup or player join
```
------------------------------------[ 17.04.2021 18:21:58 ]-----------------------------------
ChatControlRed 10.3.5 encountered a NoSuchMethodError
Running Paper 1.15.2-R0.1-SNAPSHOT and Java 11.0.9.1
Plugins: LuckPerms v5.1.93, WorldEdit v7.1.0;8e55131, SAML v2.0.0, PlaceholderAPI v2.10.9, WorldBorder v2.0.9, NametagEdit v4.4.10, Multiverse-Core v4.1.0-b775, WorldGuard v7.0.2-SNAPSHOT;bf2593e, ChatBrawl v1.7.2, VoxelSniper v6.1.2, UltraBar v2.2.2, EpicWorldGenerator v8.1.28, ViaVersion v3.2.1, NashornPlus v1.0.0, Vault v1.7.3-b131, MiniaturePets v2.5.4, WorldEditSUI v1.7.1, Vehicles v11.5, Lands v4.9.3, Jobs v4.16.1, ProtocolLib v4.6.1-SNAPSHOT-b499, Citizens v2.0.27-SNAPSHOT (build 2100), XieAPI v0.1, Quests v3.9.5-b130, DeluxeMenus v1.13.3-Release, citizens-integration v1.0, XieCore v${buildnumber}, XieDrop v2.1, XieTokenShop v1.0.b, XieTools v2.1, ChatControlRed v10.3.5, Boss v3.10.3, XieAddons v2.1, XiePlotTab v1.0, XieEconomyAddons v0.2x
----------------------------------------------------------------------------------------------
NoSuchMethodError 'void org.yaml.snakeyaml.LoaderOptions.setMaxAliasesForCollections(int)'
	 at org.mineacademy.chatcontrol.lib.settings.SimpleYaml.saveToString(SimpleYaml.java:67)
	 at org.mineacademy.chatcontrol.lib.settings.ConfigInstance.saveDo(YamlConfig.java:2145)
	 at org.mineacademy.chatcontrol.lib.settings.ConfigInstance.lambda$save0$0(YamlConfig.java:2117)
	 at org.mineacademy.chatcontrol.lib.remain.CompRunnable$SafeRunnable.run(CompRunnable.java:68)
	 at org.bukkit.craftbukkit.v1_15_R1.scheduler.CraftTask.run(CraftTask.java:99)
	 at org.bukkit.craftbukkit.v1_15_R1.scheduler.CraftAsyncTask.run(CraftAsyncTask.java:54)
	 at com.destroystokyo.paper.ServerSchedulerReportingWrapper.run(ServerSchedulerReportingWrapper.java:22)
	 at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	 at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	 at java.base/java.lang.Thread.run(Thread.java:834)
----------------------------------------------------------------------------------------------
```
/chc debug console output 
```
>.... [18:25:57 INFO]: XaaR issued server command: /chc debug
>.... [18:25:57 INFO]: [ChatControlRed] ChatControlRed 10.3.5 encountered a NoSuchMethodError! Please check your error.log and report this issue with the information in that file.
>.... [18:25:57 INFO]: java.lang.NoSuchMethodError: 'void org.yaml.snakeyaml.LoaderOptions.setMaxAliasesForCollections(int)'
>.... [18:25:57 INFO]: at org.mineacademy.chatcontrol.lib.settings.SimpleYaml.saveToString(SimpleYaml.java:67)
>.... [18:25:57 INFO]: at org.mineacademy.chatcontrol.lib.settings.SimpleYaml.saveToString(SimpleYaml.java:58)
>.... [18:25:57 INFO]: at org.mineacademy.chatcontrol.lib.command.DebugCommand.copyFilesToDebug(DebugCommand.java:93)
>.... [18:25:57 INFO]: at org.mineacademy.chatcontrol.lib.command.DebugCommand.onCommand(DebugCommand.java:43)
>.... [18:25:57 INFO]: at org.mineacademy.chatcontrol.lib.command.SimpleCommand.execute(SimpleCommand.java:390)
>.... [18:25:57 INFO]: at org.mineacademy.chatcontrol.lib.command.SimpleCommandGroup$MainCommand.onCommand(SimpleCommandGroup.java:374)
>.... [18:25:57 INFO]: at org.mineacademy.chatcontrol.lib.command.SimpleCommand.execute(SimpleCommand.java:390)
>.... [18:25:57 INFO]: [ChatControlRed] !-----------------------------------------------------!
>.... [18:25:57 INFO]: [ChatControlRed] Failed to execute command /chc debug
>.... [18:25:57 INFO]: [ChatControlRed] !-----------------------------------------------------!
>.... [18:25:57 INFO]: [ChatControlRed 10.3.5] Command /chc debug took 11.41 ms
```
Log from error.log
```
------------------------------------[ 17.04.2021 18:25:57 ]-----------------------------------
ChatControlRed 10.3.5 encountered a NoSuchMethodError
Running Paper 1.15.2-R0.1-SNAPSHOT and Java 11.0.9.1
Plugins: LuckPerms v5.1.93, WorldEdit v7.1.0;8e55131, SAML v2.0.0, PlaceholderAPI v2.10.9, WorldBorder v2.0.9, NametagEdit v4.4.10, Multiverse-Core v4.1.0-b775, WorldGuard v7.0.2-SNAPSHOT;bf2593e, ChatBrawl v1.7.2, VoxelSniper v6.1.2, UltraBar v2.2.2, EpicWorldGenerator v8.1.28, ViaVersion v3.2.1, NashornPlus v1.0.0, Vault v1.7.3-b131, MiniaturePets v2.5.4, WorldEditSUI v1.7.1, Vehicles v11.5, Lands v4.9.3, Jobs v4.16.1, ProtocolLib v4.6.1-SNAPSHOT-b499, Citizens v2.0.27-SNAPSHOT (build 2100), XieAPI v0.1, Quests v3.9.5-b130, DeluxeMenus v1.13.3-Release, citizens-integration v1.0, XieCore v${buildnumber}, XieDrop v2.1, XieTokenShop v1.0.b, XieTools v2.1, ChatControlRed v10.3.5, Boss v3.10.3, XieAddons v2.1, XiePlotTab v1.0, XieEconomyAddons v0.2x
----------------------------------------------------------------------------------------------

More Information: 
Failed to execute command /chc debug 
NoSuchMethodError 'void org.yaml.snakeyaml.LoaderOptions.setMaxAliasesForCollections(int)'
	 at org.mineacademy.chatcontrol.lib.settings.SimpleYaml.saveToString(SimpleYaml.java:67)
	 at org.mineacademy.chatcontrol.lib.settings.SimpleYaml.saveToString(SimpleYaml.java:58)
	 at org.bukkit.configuration.file.FileConfiguration.save(FileConfiguration.java:65)
	 at org.mineacademy.chatcontrol.lib.command.DebugCommand.copyFilesToDebug(DebugCommand.java:93)
	 at org.mineacademy.chatcontrol.lib.command.DebugCommand.onCommand(DebugCommand.java:43)
	 at org.mineacademy.chatcontrol.lib.command.SimpleCommand.execute(SimpleCommand.java:390)
	 at org.mineacademy.chatcontrol.lib.command.SimpleCommandGroup$MainCommand.onCommand(SimpleCommandGroup.java:374)
	 at org.mineacademy.chatcontrol.lib.command.SimpleCommand.execute(SimpleCommand.java:390)
	 at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:159)
	 at org.bukkit.craftbukkit.v1_15_R1.CraftServer.dispatchCommand(CraftServer.java:761)
----------------------------------------------------------------------------------------------
```
Even on clear instalation of chc. Plugin seems "functional" but this error bothers me.
Edit: "Same" thing with newest version of Boss. Latest protocollib build doesn't fix this.

**Comment by ElBananaa** [comment]
> Minecraft Version: 1.15.2
> MySQL: false
> BungeeCord: false
> 
> Hi. I'm getting errors like [this](https://pastebin.com/cyf1H9jf)
> Even on clear instalation of chc. Plugin seems "functional" but this error bothers me.

Hey!
We can't access to the paste link, it says it's private. Also, please provide us your `/chc debug` output file, and finally, make sure your plugins and their dependencies are up-to-date.
If you wish to keep your configurations private, please send it to matej@mineacademy.org with the link to your issue, but keep in mind that in this case, I won't be able to provide you any help. ;)

EDIT: Also quoting this a potentially related to #898 since NoSuchMethodError was already reported on their. Waiting for the author to update their error link.
Please tag me in the comment when the link is updated, so I don't forget to take a look at your issue :)

**Comment by XaaREl** [comment]
My bad with pastebin. Updated with additional logs and info i guess :)
@ElBananaa

**Comment by ElBananaa** [comment]
> My bad with pastebin. Updated with additional logs and info i guess :)
> @ElBananaa

Hey!
Thanks.
Could you please give me your exact paper version (I mean, the build version)? I'll try to reproduce it with this version since I can't on the latest version of paper 1.16.5.

**Comment by XaaREl** [comment]
I'm using latest version for 1.15.2, exactly 391

**Comment by ElBananaa** [comment]
Hey!
So after running a few tests, I can confirm this issue happens on every version I tried which is under 1.16.5 (I tried 1.15.2, 1.14.4, 1.13.2 & 1.12.2).
The tests were made under a completely fresh install, the only command I ran was the command to complete the tour. I then tried to restart the server, but the issue persists.
This is probably a plugin-side issue. Let's wait for @kangarko regarding this ;)
